datafusion/core/tests/physical_optimizer/projection_pushdown.rs

```diff
@@ -89,6 +92,10 @@ impl ScalarUDFImpl for DummyUDF {
     fn return_type(&self, _arg_types: &[DataType]) -> Result<DataType> {
         Ok(DataType::Int32)
     }
+
+    fn invoke_with_args(&self, _args: ScalarFunctionArgs) -> Result<ColumnarValue> {
+        panic!("dummy - not implemented")
```

Comment:
what about use `not_impl_err`

Reply:
I just copied https://github.com/apache/datafusion/pull/15123/files#diff-351499880963d6a383c92e156e75019cd9ce33107724a9635853d7d4cd1898d0R1602 - but happy to change, would you prefer `not_impl_err`?

Reply:
I checked other traits, it would prefer `unimplemented!()`.

Reply:
FWIW this is a test